mirgecom/multiphysics/phenolics.py

```diff
@@ -0,0 +1,83 @@
+""":mod:`mirgecom.multiphysics.phenolics` handles phenolics modeling."""
+
+__copyright__ = """
+Copyright (C) 2023 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from meshmode.dof_array import DOFArray
+from pytools.obj_array import make_obj_array
+from mirgecom.fluid import ConservedVars
+
+
+def initializer(dcoll, gas_model, material_densities, temperature,
+                gas_density=None, pressure=None):
```

Comment:
I would either: 1) add an optional `dd=DD_VOLUME_ALL` argument (even though it's not currently used), or 2) remove the `dcoll` and pass `dim` directly instead.